FilmlisteParser.js

```diff
@@ -1,138 +1,160 @@
-                                url_website: parsed[9],
-                                url_subtitle: parsed[10],
-                                url_video: parsed[8],
-                                url_video_low: createUrlFromBase(parsed[8], parsed[12]),
-                                url_video_hd: createUrlFromBase(parsed[8], parsed[14])
-                            };
-
-                            buffer.push(['sadd', setKey, JSON.stringify(entry)]);
-                            entries++;
-
-                            if (currentLine % 500 == 0) {
-                                redis.batch(buffer).exec();
-                                buffer = [];
-                                ipc.send('state', {
-                                    entries: entries,
-                                    progress: getProgress()
-                                });
-                            }
-                        } else if (last) {
-                            redis.batch(buffer).exec(() => {
-                                ipc.send('state', {
-                                    entries: entries,
-                                    progress: 1
-                                });
-                                ipc.send('done');
-
-                                fs.close(fd, () => {
-                                    setTimeout(() => process.exit(0), 500);
-                                });
-                            });
-                        }
-
-                        getNext();
-                    });
-                }
+  fs.open(file, 'r', (err, fd) => {
+    if (err) {
+      handleError(err);
+      return;
+    }
+
+    fs.fstat(fd, (err, stats) => {
+      if (err) {
+        handleError(err);
+        return;
+      }
+
+      let currentChannel,
+          currentTopic,
+          buffer = [],
+          currentLine = 0;
+
+      const filesize = stats.size,
+            fileStream = fs.createReadStream(null, {fd: fd, autoClose: true}),
+            lineReaderSeparator = {separator: /^{"Filmliste":|,"X":|}$/}
+
+      const getProgress = () => {
+        return fileStream.bytesRead / filesize;
+      };
+
+      lineReader.eachLine(fileStream, lineReaderSeparator, (line, last, getNext) => {
+        currentLine++;
+        if (currentLine === 1) {
+          return getNext();
+        }
+        if (currentLine === 2) {
+          redis.set(timestampKey, handleListMeta({line}));
+          return getNext();
+        }
+
+        if (last) {
+          return redis.batch(buffer).exec(() => {
+            ipc.send('state', {
+              entries: currentLine - 2,
+              progress: 1
             });
+            ipc.send('done');
+            fs.close(fd, () => setTimeout(() => process.exit(0), 500));
+          });
         }
-    });
 
+        [currentChannel, currentTopic, entry] = mapListLineToRedis({line, currentChannel, currentTopic})
```

Comment:
But the `currentChannel` and the `currentTopic` thingie have to be not newly defined, as they'll carry on the topic or the channel because this JSON where parsing isn't really JSON.
So we're either getting them refreshed from the line, or they'll have to stay the same until a line with a new channel or topic appears. 

It's rather a strange slang, that could be considered a crossing between CSV and JSON and key omitting. 
What you suggest would break its functionality. 
With this `mapListLineToRedis()` function I intended to make the line iterator more explicit... so externalise the line processing and get in return a shorter iterator where one can see the end and it's beginning.

Actually, I would prefer to get real JSON with nesting objects as a Filmliste thingie because that would not only be easier to handle. It would even use up fewer bytes in the list, which is what I guess the authors of that Filmliste thing intended when they omitted the keys in the first place.